### PR TITLE
Remove SearchView component

### PR DESCRIPTION
**Pull request by yeexinc**
[DYN-852](https://jira.autodesk.com/browse/DYN-852)

- Remove `SearchView` component and move the relevant states (`structured`, `detailed`, `selectedCategories` etc. to `LibraryContainer`).
- Move the `SearchBar` component to `LibraryContainer` so that it exists at the same level as the library items
- Implement a `Searcher` component that contains the utility functions for searching

### CODE REVIEW

File: src/components/SearchView.tsx [hunk 1]
```diff
@@ -1,172 +0,0 @@
-import * as React from "react";
```
Comment (Benglin) [inline src/components/SearchView.tsx]: @fanwgwg [has changed something](https://github.com/DynamoDS/librarie.js/pull/42/files) in `SearchView` so please make sure those changes persist over here, too. @fanwgwg can you help review this when YX makes changes? Thanks!

File: src/Searcher.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+interface SearcherProps {
+    onSearchModeChanged: SearchModeChangedFunc;
+    libraryContainer: LibraryContainer;
+    sections: ItemData[];
+    categories: string[];
+}
+
+interface SearcherStates {}
+
+export class Searcher extends React.Component<SearcherProps, SearcherStates> {
```
Comment (Benglin) [inline src/Searcher.tsx]: `Searcher` has no reason to be a `React.Component`, it should be a regular TypeScript class so unit test cases can be written for it with mocha/chai frameworks.

File: src/components/LibraryContainer.tsx [hunk 3]
```diff
@@ -20,25 +23,53 @@ export interface LibraryContainerProps {
...
+        })
+
+        this.state = {
+            inSearchMode: false,
+            searchText: '',
+            displayMode: "list",
+            selectedCategories: this.searchCategories,
+            structured: false,
+            detailed: false
+        };
```
Comment (Benglin) [inline src/components/LibraryContainer.tsx]: There are [many changes for this file](https://github.com/DynamoDS/librarie.js/pull/61/files) also, I think the best way to do things would be closing this PR, and start the work again based on the latest code. Does that make sense?

**Comment by Benglin** [comment]
Hi @yeexinc, I think to incorporate the latest changes from everyone, this is best closed and redone based on the latest code base. That may make it easier for you, does it make sense?

**Comment by yeexinc** [comment]
@Benglin - Sure thing, I'll close this and make a separate one.

**Pull request closed without merging**